### PR TITLE
Refined create_vm method in Softlayer VM package

### PR DESCRIPTION
**Pull request by jianqiu**
In this PR, the `create_vm` action had been refined, moved common methods, such like `CreateVirtualGuestTemplate`, `CreateAgentMetadata`, `CreateDisksSpec`, `AppendPowerDNSToNetworks`, `Base64EncodeData`, `TimeStampForTime` into `vm_utils.go` in softlayer vm package.

### CODE REVIEW

File: softlayer/vm/network.go [hunk 1]
```diff
@@ -24,3 +24,8 @@ func (ns Networks) First() Network {
 }
 
 func (n Network) IsDynamic() bool { return n.Type == "dynamic" }
+
+func (n Network) AppendDNS(addr string) Network {
```
Comment (maximilien) [inline softlayer/vm/network.go]: Where is the test for this method? Is it still needed? If not remove it, if yes, then add a test.

**Comment by maximilien** [comment]
One minor comment @jianqiu, please address soon so I can merge.

Looking good.

**Comment by jianqiu** [comment]
@maximilien  A new unit tests for `network.go` had been created, please review the code again, thanks!

**Comment by jianqiu** [comment]
@maximilien  A new test `network_test.go` is covering test cases for all the public methods in `network.go`

**Comment by maximilien** [comment]
Cool. Merging. Cheers :beers:

**Pull request merged**